src/components/ui/Swap/SwapHome.tsx

```diff
@@ -98,47 +88,29 @@ export const SwapHome = ({ onSetting }: SwapHomeProps) => {
   };
 
   useEffect(() => {
-    setInterval(() => refetchBalance(), 5000);
+    const intervalId = setInterval(() => refetchBalance(), 5000);
+    
     return () => {
       clearRouteChecking();
+      clearInterval(intervalId)
     };
   }, []);
 
-  useEffect(() => {
-    if (debouncedInputValue == 0) {
-      setRouteData(undefined);
+  const {
+    isLoading:fetchLoading, 
+    isError:fetchError,
+  } = useQuery({
+    queryKey: [inputTokenIndex, outputTokenIndex, debouncedInputValue],
+    refetchInterval: 5000,
+    queryFn: async () => {
+      if (debouncedInputValue == 0) {
+        setRouteData(undefined);
+        clearRouteChecking();
```

Comment:
We do not use interval anymore so `clearRouteChecking` function is obsolete. Also, we can remove `intervalId` variable